CommonTools/Utils/src/cutParser.cc

```diff
@@ -17,6 +17,9 @@ bool reco::parser::cutParser(const edm::TypeWithDict& t, const std::string& cut,
     return true;
   } else {
     using namespace boost::spirit::classic;
+    auto previous = gInterpreter->SetClassAutoparsing(true);
+    auto setPrevious = [previous](TInterpreter* interpreter) { interpreter->SetClassAutoparsing(previous); };
+    std::unique_ptr<TInterpreter, decltype(setPrevious)> guard(gInterpreter, setPrevious);
```

Comment:
I _hope_ the `reco::parser::cutParser()` (or `StringCutObjectSelector` constructor` is called only from module constructors, because adjusting this global setting in a parallel section is not "logically thread safe".